bsc/api_basic.go

```diff
@@ -0,0 +1,171 @@
+package bsc
+
+import (
+	"context"
+	"log"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/bnb-chain/greenfield-go-sdk/bsctypes"
+	bsccommon "github.com/bnb-chain/greenfield-go-sdk/common"
+)
+
+type IBasicClient interface {
+	SendTx(ctx context.Context, nonce uint64, toAddr *common.Address, amount *big.Int, gasPrice *big.Int, data []byte) (*common.Hash, error)
+	GetDeployment() *bsctypes.Deployment
+	GetMinAckRelayFee(ctx context.Context) (relayFee *big.Int, minAckRelayFee *big.Int, err error)
+	GetCallbackGasPrice(ctx context.Context) (gasPrice *big.Int, err error)
+	CheckTxStatus(ctx context.Context, tx *common.Hash) (bool, error)
+}
+
+func (c *Client) GetDeployment() *bsctypes.Deployment {
+	return c.deployment
+}
+
+func (c *Client) SendTx(ctx context.Context, nonce uint64, toAddr *common.Address, amount *big.Int, gasPrice *big.Int, data []byte) (*common.Hash, error) {
+	if nonce == 0 {
+		n, err := c.chainClient.PendingNonceAt(ctx, *c.defaultAccount.GetAddress())
+		if err != nil {
+			return nil, err
+		}
+		nonce = n
+	}
+	gasLimit := uint64(5e6) // Assuming the gas limit is static, adjust as necessary
+	var err error
+	if gasPrice == nil {
+		gasPrice, err = c.chainClient.SuggestGasPrice(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Create the transaction using NewTx instead of the deprecated NewTransaction
+	tx := types.NewTx(&types.LegacyTx{
+		Nonce:    nonce,
+		To:       toAddr,
+		Value:    amount,
+		Gas:      gasLimit,
+		GasPrice: gasPrice,
+		Data:     data,
+	})
+
+	chainId, err := c.chainClient.ChainID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(chainId), c.defaultAccount.GetKeyManager().GetPrivateKey())
```

Comment:
We'd better use `types.NewLondonSigner` now

Reply:
updated